### DISCUSSION TITLE
How is josh different from partial clone and sparse-checkout?

### DISCUSSION THREAD

**Discussion by jjangga0214** [josh-project/josh#421] [Q&A] [answered]
Hi, I just got to know the project.

How is this different from, or working together with, [partial clone](https://git-scm.com/docs/partial-clone) and `sparse-checkout`(See `git sparse-checkout --help`)?

Thanks.

**Comment by christian-schilling** [comment]
Not really an issue.

**Comment by robotrapta** [comment]
Okay but could you maybe answer the question anyway?

**Comment by christian-schilling** [comment, marked as answer]
Different from sparse-checkout:
Josh operates on the git object graph and is unrelated to checking out files and the working tree on the filesystem, which is the only thing sparse-checkout is concerned with. A sparse checkout does not influence the contents of the object database and also not what gets downloaded over the network.
Both can certainly be used together if needed.

Partial clone:
A partial clone will cause git to download only parts of an object database according to some predicate. It is still the same object database with the history having the same commits and sha1s. It still allows loading skipped parts of the object database at a later point.
Josh creates an alternate history that has no reference to the skipped parts. It is as such very similar to git filter-branch just faster, with added features and a different user interface.
